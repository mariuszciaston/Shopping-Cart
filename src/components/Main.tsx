const Main = () => {
  return (
    <main
      className="
    
    flex-1 bg-pink-100"
    >
      MAIN
    </main>
  );
};

export default Main;
