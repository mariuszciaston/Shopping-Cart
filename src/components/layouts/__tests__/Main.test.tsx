import { render, screen } from "@testing-library/react";
import Main from "../Main";
import { describe, it, expect } from "vitest";
import "@testing-library/jest-dom";

describe("Main component", () => {
  it("renders children", () => {
    render(
      <Main>
        <div>Test</div>
      </Main>,
    );
    const child = screen.getByText("Test");
    expect(child).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <Main>
        <p>children test</p>
      </Main>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
