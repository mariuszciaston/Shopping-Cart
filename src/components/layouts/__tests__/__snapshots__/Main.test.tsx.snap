// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main component > matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="flex-1 bg-gray-100"
  >
    <div
      class="mx-auto grid h-full w-full max-w-[1280px] content-start py-8 sm:p-8"
    >
      <p>
        children test
      </p>
    </div>
  </main>
</DocumentFragment>
`;
